src/charm.py

```diff
@@ -213,6 +245,13 @@ def _on_update_status(self, _):
             )
             return
 
+        self.upgrade._reconcile_upgrade(event)
+        if self.unit.status in (
+            Config.Status.UNHEALTHY_UPGRADE,
+            Config.Status.INCOMPATIBLE_UPGRADE,
+        ):
+            return
```

Comment:
I *think* we might want to have `self.update_tls_sans()` before this. `self.update_tls_sans()` does some self healing and it could be the reason that there is an unhealthy upgrade?

Reply:
Oh sure

Reply:
502f904